### PR TITLE
NAS-123782 / 24.04 / Update default recordsize for draid pool based datasets/zvols

### PR DESCRIPTION
**Pull request by Qubad786**
## Context

A request was made to redefine default values and minimum values for recordsize/volblocksize attrs of a dataest/zvol for draid based pools.

### CODE REVIEW

**Comment by yocalebo** [comment]
@truenas/middleware ping

**Review by themylogin — APPROVED** [review]

**Comment by yocalebo** [comment]
retest this please

File: tests/api2/test_draid_record_and_block_size.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+                'type': pool_type,
+                **({'draid_data_disks': 1} if pool_type.startswith('DRAID') else {})
+            }],
+        },
+        'allow_duplicate_serials': True,
+    }) as pool_name:
+        assert call('pool.dataset.get_instance', pool_name['name'])['recordsize']['value'] == record_size
+
+
+@pytest.mark.parametrize(
```
Comment (themylogin) [inline tests/api2/test_draid_record_and_block_size.py:36]: We will have a relatively large number of pool creation/deletion operations here, running these tests might take up a very long time. Can we test the complete recordsize logic with an integration test and here only test a part of it?

**Review by themylogin — CHANGES_REQUESTED** [review]

File: tests/api2/test_draid_record_and_block_size.py (hunk 1 above)
Comment (Qubad786) [reply tests/api2/test_draid_record_and_block_size.py:36]: I'll use a fixture here so we avoid creating a pool each time these tests are run and that should solve this problem ✅

Comment (yocalebo) [reply tests/api2/test_draid_record_and_block_size.py:36]: Agree with @themylogin's review. When you use a fixture, make sure that the pool is cleaned up after this module is run. Thanks!

**Review by themylogin — APPROVED** [review]

**Pull request merged**